### ISSUE TITLE
Constraint violation if upgraded from v5 to v6 when perform formService.submitTaskFormData()

### ISSUE THREAD

**Issue by quaff** [Activiti/Activiti#1217]
```
### Error updating database.  Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Cannot delete or update a parent row: a foreign key constraint fails (`ironrhino_activiti`.`act_ru_event_subscr`, CONSTRAINT `ACT_FK_EVENT_EXEC` FOREIGN KEY (`EXECUTION_ID_`) REFERENCES `ACT_RU_EXECUTION` (`ID_`))
### The error may involve org.activiti.engine.impl.persistence.entity.ExecutionEntityImpl.deleteExecution-Inline
### The error occurred while setting parameters
### SQL: delete from ACT_RU_EXECUTION where ID_ = ? and REV_ = ?
### Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Cannot delete or update a parent row: a foreign key constraint fails (`ironrhino_activiti`.`act_ru_event_subscr`, CONSTRAINT `ACT_FK_EVENT_EXEC` FOREIGN KEY (`EXECUTION_ID_`) REFERENCES `ACT_RU_EXECUTION` (`ID_`))
```

```
### Error updating database.  Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Cannot delete or update a parent row: a foreign key constraint fails (`ironrhino_activiti`.`act_ru_task`, CONSTRAINT `ACT_FK_TASK_EXE` FOREIGN KEY (`EXECUTION_ID_`) REFERENCES `ACT_RU_EXECUTION` (`ID_`))
### The error may involve org.activiti.engine.impl.persistence.entity.ExecutionEntityImpl.deleteExecution-Inline
### The error occurred while setting parameters
### SQL: delete from ACT_RU_EXECUTION where ID_ = ? and REV_ = ?
### Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Cannot delete or update a parent row: a foreign key constraint fails (`ironrhino_activiti`.`act_ru_task`, CONSTRAINT `ACT_FK_TASK_EXE` FOREIGN KEY (`EXECUTION_ID_`) REFERENCES `ACT_RU_EXECUTION` (`ID_`))
```
Activiti should postpone deleteExecution, or drop constraint like
```
ALTER TABLE ACT_RU_EVENT_SUBSCR DROP FOREIGN KEY ACT_FK_EVENT_EXEC ;

ALTER TABLE ACT_RU_TASK DROP FOREIGN KEY ACT_FK_TASK_EXE ;
```

**Comment by quaff** [comment]
BTW, I'm not using activiti5-compatibility

**Comment by Salaboy** [comment]
@quaff are you still facing this issue? I'm closing it due to inactivity. If you are still blocked by this please get in touch and reopen this issue so we can fix it or improve this for the next version of the framework.

**Comment by quaff** [comment]
I have turn to flowable

**Comment by Salaboy** [comment]
@quaff good stuff! enjoy Open Source!

**Comment by Salaboy** [comment]
@ravenblackdusk which version are you using exactly? I will reopen if it is one of the latest versions.

**Comment by balsarori** [comment]
I ran the following test against MySQL DB without issues

https://github.com/Activiti/Activiti/blob/6.x/modules/activiti-engine/src/test/java/org/activiti/engine/test/api/form/FormServiceTest.java#L423

@ravenblackdusk can you provide steps to reproduce or maybe a unit test to use in order reproduce the issue?

**Comment by ravenblackdusk** [comment]
@Salaboy 6.0.0
@balsarori I'm on it

**Comment by ravenblackdusk** [comment]
writing the test case proved harder than expected. I've written a springBootTest as such:

```
@RunWith(SpringJUnit4ClassRunner.class)
@SpringBootTest(classes = TestConfiguration.class)
public class CompensationTest {
    @Autowired
    private RepositoryService repositoryService;
    @Autowired
    private RuntimeService runtimeService;

    @Test
    public void testCompensation() {
        final Deployment deployment = repositoryService.createDeployment()
                .addBpmnModel("new.bpmn2.xml", createCompTestBpmnModel()).deploy();
        System.out.println(repositoryService.createProcessDefinitionQuery().list().size());
    }

    public BpmnModel createCompTestBpmnModel() {
        final BpmnModel model = new BpmnModel();
        final Process process = new Process();
        model.addProcess(process);
        process.setId("comp");
        process.setName("comp");

        final StartEvent startEvent = new StartEvent();
        startEvent.setId("start");
        process.addFlowElement(startEvent);

        final UserTask userTask = new UserTask();
        userTask.setName("The First Task");
        userTask.setId("task1");
        userTask.setAssignee("kermit");
        process.addFlowElement(userTask);

        final BoundaryEvent boundaryEvent = new BoundaryEvent();
        boundaryEvent.setId("boundary");
        boundaryEvent.setAttachedToRefId("task1");
        boundaryEvent.addEventDefinition(new CompensateEventDefinition());
        process.addFlowElement(boundaryEvent);

        final UserTask userTask2 = new UserTask();
        userTask2.setName("The Second Task");
        userTask2.setId("task2");
        userTask2.setAssignee("kermit");
        process.addFlowElement(userTask2);

        final ThrowEvent throwEvent = new ThrowEvent();
        throwEvent.setId("throw");
        throwEvent.setName("throw");
        throwEvent.addEventDefinition(new CompensateEventDefinition());
        process.addFlowElement(throwEvent);

        final UserTask userTask3 = new UserTask();
        userTask3.setName("The Third Task");
        userTask3.setId("task3");
        userTask3.setAssignee("kermit");
        process.addFlowElement(userTask3);

        final UserTask compTask = new UserTask();
        compTask.setName("The comp Task");
        compTask.setId("comp");
        compTask.setAssignee("kermit");
        compTask.setForCompensation(true);
        process.addFlowElement(compTask);

        process.addFlowElement(new SequenceFlow("start", "task1"));
        process.addFlowElement(new SequenceFlow("task1", "task2"));
        process.addFlowElement(new SequenceFlow("task2", "throw"));
        process.addFlowElement(new SequenceFlow("throw", "task3"));
        process.addFlowElement(new SequenceFlow("boundary", "comp"));

        return model;
    }
}
```
the deployment is successfully added to DB but the process definition list is empty (perhaps some configuration problem). anyways if you manage to run such a process; when task2 is submitted and compensation is thrown; if you then get the active activities of the process instance:

`runtimeService.getActiveActivityIds(processDefinitionId);`

both task3 and comp is returned(which i think task3 should become active after comp is submitted).
now if you try to run task3 the above mentioned DB error is thrown. but if you submit comp first and then task3, no problem will occur.
